Hexastore.Rocks/TripleExtensions.cs

```diff
@@ -39,37 +40,56 @@ public static byte[] ToBytes(this Triple triple)
         public static Triple ToTriple(this byte[] spoBytes)
         {
             var index = 0;
-            var sBytes = ReadNext(spoBytes, index);
-            index += 4 + sBytes.Length;
-            var pBytes = ReadNext(spoBytes, index);
-            index += 4 + pBytes.Length;
-            var arrayIndexBytes = ReadNext(spoBytes, index);
-            index += 4 + arrayIndexBytes.Length;
-            var idBytes = ReadNext(spoBytes, index);
-            index += 4 + idBytes.Length;
-            var oTypeBytes = ReadNext(spoBytes, index);
-            index += 4 + oTypeBytes.Length;
-            var oBytes = ReadNext(spoBytes, index);
+            var (s, slen) = ReadString(spoBytes, index);
+            index += 4 + slen;
```

Comment:
nit: there are a lot of 4's in these methods and I don't really understand their significance. Perhaps on the next iteration we can move them to an aptly named variable to help clarify